src/java/org/apache/cassandra/db/virtual/AbstractWritableVirtualTable.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    protected void applyColumnDelete(DecoratedKey partitionKey, ClusteringPrefix<?> clusteringColumns, Cell<?> cell)
+    {
+        throw new InvalidRequestException("Column deletion is not supported by table " + metadata);
+    }
+
+    protected abstract void applyColumnUpdate(DecoratedKey partitionKey, ClusteringPrefix<?> clusteringColumns, Cell<?> cell);
+
+    public static abstract class SimpleWritableVirtualTable extends AbstractWritableVirtualTable {
+
+        protected SimpleWritableVirtualTable(TableMetadata metadata)
+        {
+            super(metadata);
+        }
+
+        @Override
+        protected void applyPartitionDelete(DecoratedKey partitionKey)
+        {
+            applyPartitionDelete(extractPartitionKeyColumnValues(partitionKey));
+        }
+
+        protected void applyPartitionDelete(Object[] partitionKeyColumnValues)
+        {
+            throw new InvalidRequestException("Partition deletion is not supported by table " + metadata);
+
+        }
+
+        @Override
+        protected void applyRangeTombstone(DecoratedKey partitionKey, Slice slice)
+        {
+            ClusteringBound<?> startClusteringColumns = slice.start();
+            Object[] startClusteringColumnValues = extractClusteringColumnValues(startClusteringColumns);
+
+            ClusteringBound<?> endClusteringColumns = slice.end();
+            Object[] endClusteringColumnValues = extractClusteringColumnValues(endClusteringColumns);
+
+            // It is a prefix of clustering columns that have equal condition. For example, if there are two clustering
+            // columns c1 and c2, then it will have c1. In case of a single clustering column the prefix is empty.
+            int clusteringColumnsPrefixLength = Math.max(startClusteringColumnValues.length, endClusteringColumnValues.length) - 1;
+            Object[] clusteringColumnValuesPrefix = new Object[clusteringColumnsPrefixLength];
+            System.arraycopy(startClusteringColumnValues, 0, clusteringColumnValuesPrefix, 0, clusteringColumnsPrefixLength);
+
+            Object startClusteringColumnValue = startClusteringColumns.isBottom()
+                    ? null : startClusteringColumnValues[startClusteringColumnValues.length - 1];
+            boolean isStartClusteringColumnInclusive = startClusteringColumns.isInclusive();
+
+            Object endClusteringColumnValue = endClusteringColumns.isBottom()
+                    ? null : endClusteringColumnValues[endClusteringColumnValues.length - 1];
+            boolean isEndClusteringColumnInclusive = endClusteringColumns.isInclusive();
+
+            applyRangeTombstone(extractPartitionKeyColumnValues(partitionKey),
+                    clusteringColumnValuesPrefix,
+                    startClusteringColumnValue,
+                    isStartClusteringColumnInclusive,
+                    endClusteringColumnValue,
+                    isEndClusteringColumnInclusive);
```

Comment:
I like `Range` idea! Done.

Regarding the prefix, the idea is like that:
All clustering columns except the last level should have equality condition and range comparison is not applicable to them. For example: `c1='c1_1' AND c2='c2_1' AND c3>'c3_1'` it is impossible to specify `c1>='c1_1' AND c2='c2_1' AND c3>'c3_1'` or `c1='c1_1' AND c2<'c2_1' AND c3>'c3_1'`. So basically prefix contains value for all "first" clustering columns. For this example it would be ["c1_1", "c2_1"].

It was mentioned in the comment for the prefix calculation. Now I moved these details to the javadoc for `applyRangeTombstone`.